src/js/themes/base.d.ts

```diff
@@ -195,7 +196,13 @@ interface ButtonType {
         secondary?: ButtonKindType;
       })
     | { [key: string]: ButtonKindType };
-  disabled?: ButtonKindType & { opacity?: OpacityType };
+  disabled?:
+    | (ButtonKindType & {
+        default?: ButtonKindType;
+        primary?: ButtonKindType;
+        secondary?: ButtonKindType;
+      })
+    | ({ [key: string]: ButtonKindType } & { opacity?: OpacityType });
```

Comment:
Could we just have this?
```suggestion
  disabled?: ({ [key: string]: ButtonKindType } & { opacity?: OpacityType });
```

Reply:
This structure mimics the current structure for `active` and `hover`.

Reply:
Ok I'm fine moving forward with this then